src/value/owned.rs

```diff
@@ -569,13 +601,55 @@ mod value_tests {
             &|e: &OwnedElement| {
                 assert_eq!(Some(&OwnedStruct::from_iter(vec![("greetings", OwnedElement::from(OwnedValue::String("hello".into())))].into_iter())), e.as_struct());
             }
-        )
+        ),
+        // TODO consider factoring this out of the value tests to make it more contained
+        // TODO consider adding non-equivs for this (really symbol token tests are probably better for this)
+        eq_annotations => [
```

Comment:
I'm stumbling on this code block, likely because I'm new to `rstest`. This is called `eq_annotations`, but with the exception of the leading empty vec pair example all of the token vectors below contain non-equivalent tokens.

Could you walk me through this a bit?

Reply:
This `rs_test`-ism makes a test case for every test case with each top-level `Vec`.

Each `Vec` within that is a list of annotations to apply and check element equality for.

```rust
        eq_annotations => [
            // CASE 1
            vec![vec![], vec![]],

            // CASE 2
            vec![
                vec![text_token("hello"), text_token("world")],
                // containing local sids only
                vec![
                    text_token("hello").with_local_sid(20),
                    local_sid_token(21).with_text("world"),
                ],
                // mix of local sid, but all with sources
                vec![
                    text_token("hello").with_source("hello_table", 2),
                    text_token("world").with_local_sid(59).with_source("world_table", 200)
                ]
            ],

            // CASE 3
            vec![
                vec![
                    // local sid only with no text are all $0 equivalent
                    local_sid_token(21),
                    // source import table is the comparator for unknown cases
                    local_sid_token(22).with_source("hello_table", 2)
                ],
                vec![
                    local_sid_token(0),
                    local_sid_token(400).with_source("hello_table", 2),
                ],
                vec![
                    local_sid_token(200),
                    text_token("hello").with_source("hello_table", 2),
                ],
                vec![
                    local_sid_token(99),
                    text_token("hello").with_local_sid(500).with_source("hello_table", 2),
                ],
            ],
        ]
```